### PR TITLE
cli: Add KV `export` and `import`

### PR DESCRIPTION
**Pull request by jen20**
This commit adds two new commands to the Consul KV CLI, which export and import a JSON formatted representation of the Consul KV tree. It is useful to migrate parts of the KV tree between unrelated Consul clusters, and could also be used for initial data population of the KV store.

### CODE REVIEW

File: command/kv_export.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  and writes a JSON representation to stdout. This can be used with the command
+  "consul kv import" to move entire trees between Consul clusters.
+
+      $ consul kv export vault
+
+  For a full list of options and examples, please see the Consul documentation.
+
+` + apiOptsText + `
+
+KV Get Options:
```
Comment (slackpad) [inline command/kv_export.go]: Should say `KV Export Options:`

File: command/kv_get.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ KV Get Options:
 
 func (c *KVGetCommand) Run(args []string) int {
 	cmdFlags := flag.NewFlagSet("get", flag.ContinueOnError)
-	cmdFlags.Usage = func() { c.Ui.Output(c.Help()) }
```
Comment (slackpad) [inline command/kv_get.go]: This looks like an unrelated change, but if this is implied by not setting `.Usage` then makes sense and is fine.

File: command/kv_import.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      $ cat filename.json | consul kv put config/program/license -
+
+  Alternatively the data may be provided as the final parameter to the command,
+  though care must be taken with regards to shell escaping.
+
+  For a full list of options and examples, please see the Consul documentation.
+
+` + apiOptsText + `
+
+KV Get Options:
```
Comment (slackpad) [inline command/kv_import.go]: This is stale, too.

File: command/kv_import.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  generated by the "consul kv export" command.
+
+  The data can be read from a file by prefixing the filename with the "@"
+  symbol. For example:
+
+      $ consul kv import @filename.json
+
+  Or it can be read from stdin using the "-" symbol:
+
+      $ cat filename.json | consul kv put config/program/license -
```
Comment (slackpad) [inline command/kv_import.go]: Needs an update.

File: command/kv_import.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	case 0:
+		return "", errors.New("Missing DATA argument")
+	case 1:
+	default:
+		return "", fmt.Errorf("Too many arguments (expected 1 or 2, got %d)", len(args))
+	}
+
+	data := args[0]
+
+	switch data[0] {
```
Comment (slackpad) [inline command/kv_import.go]: Will this crash if you pass an empty quoted string in as the first arg?

File: command/kv_import_test.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+	pair, _, err := client.KV().Get("foo", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if strings.TrimSpace(string(pair.Value)) != "bar" {
+		t.Fatalf("bad: expected: bar, got %s", pair.Value)
+	}
+}
```
Comment (slackpad) [inline command/kv_import_test.go]: Might as well check `foo/a` as well.

File: command/kv_get.go (hunk 2 above)
Comment (cbednarski) [inline command/kv_get.go]: Is this intended? Maybe I missed it but I didn't see this move elsewhere.

File: command/kv_import.go (hunk 5 above)
Comment (jen20) [inline command/kv_import.go]: Likely yes - this is copied from `kv put` so that probably has the same behaviour. I'll update.

**Comment by jen20** [comment]
I've fixed up most of the review comments, probably also need to look at `kv put` for the `data[0]` issue.

**Comment by slackpad** [comment]
Latest changes look good to merge. I think it's fine if you want to fix the empty arg issue here for `put` as well.

**Comment by jen20** [comment]
The panic in `kv put` is fixed in #2635, the same bug in `kv import` is fixed here. I'm currently waiting for Travis confirmation then I'll merge this one and leave #2635 for review of the new behaviour.

**Pull request merged**